daemon/cmd/state.go

```diff
@@ -322,7 +322,26 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState, endpoi
 		}
 	}
 
+	if option.Config.EnableIPSec {
+		for _, ep := range state.restored {
+			if ep.IsHost() {
+				log.WithField(logfields.EndpointID, ep.ID).Info("Successfully restored endpoint. Scheduling regeneration")
+				if err := ep.RegenerateAfterRestore(endpointsRegenerator, d.bwManager, d.fetchK8sMetadataForEndpoint); err != nil {
+					log.WithField(logfields.EndpointID, ep.ID).WithError(err).Debug("error regenerating restored host endpoint")
+					epRegenerated <- false
+				} else {
+					epRegenerated <- true
+				}
+				break
+			}
+		}
+	}
+
 	for _, ep := range state.restored {
+		if ep.IsHost() && option.Config.EnableIPSec {
+			// The host endpoint was handled above.
+			continue
+		}
```

Comment:
For bisect reasons, I think we want this patch earlier in the series.

(and the new `OVERLAY` mark now *also* requires that `bpf_host` is regenerate prior to `bpf_overlay` :slightly_smiling_face: )

Reply:
@julianwiedmann 

ACK on reorder of commit. 

Re: > and the new OVERLAY mark now also requires that bpf_host is regenerate prior to bpf_overlay 🙂

I don't think this is the case, since we will backport this change into v1.17. Therefore, `bpf_overlay` will be marking packets correctly with `MARK_MAGIC_OVERLAY_ENCRYPTED` prior to the v1.18 upgrade. Does my thinking check out there?

Reply:
>since we will backport this change into v1.17. Therefore, bpf_overlay will be marking packets correctly with MARK_MAGIC_OVERLAY_ENCRYPTED prior to the v1.18 upgrade. Does my thinking check out there?

ah yep, that should work! Do you have a v1.17 draft somewhere already?

Reply:
@julianwiedmann doing this right meow. Unless I run into any horrible conflicts, should be up pretty soon.

Reply:
Backport draft: https://github.com/cilium/cilium/pull/37993

@jschwinger233 identified one of the existing issues with Cluster Mesh.
The other two failing issues are more trivial and just appear to be unit tests items. 

I am working on hacking the ipsec upgrade tests in this PR to reference the v1.17 backport image in that linked PR. We'll run tests again with the backport v1.17 image and ensure all our downgrade leaks go away.